app/src/main/java/app/revanced/manager/ui/screen/settings/update/UpdateScreen.kt

```diff
@@ -0,0 +1,224 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Update
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Button
+import androidx.compose.material3.Divider
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.LinearProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import app.revanced.manager.BuildConfig
+import app.revanced.manager.R
+import app.revanced.manager.ui.component.AppTopBar
+import app.revanced.manager.ui.component.settings.Changelog
+import app.revanced.manager.ui.viewmodel.UpdateViewModel
+import app.revanced.manager.ui.viewmodel.UpdateViewModel.Changelog
+import app.revanced.manager.ui.viewmodel.UpdateViewModel.State
+import app.revanced.manager.util.formatNumber
+import app.revanced.manager.util.relativeTime
+import com.gigamole.composefadingedges.content.FadingEdgesContentType
+import com.gigamole.composefadingedges.content.scrollconfig.FadingEdgesScrollConfig
+import com.gigamole.composefadingedges.fill.FadingEdgesFillType
+import com.gigamole.composefadingedges.verticalFadingEdges
+import org.koin.androidx.compose.getViewModel
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+@Stable
+fun UpdateScreen(
+    onBackClick: () -> Unit,
+    vm: UpdateViewModel = getViewModel()
+) {
+    Scaffold(
+        topBar = {
+            AppTopBar(
+                title = stringResource(R.string.updates),
+                onBackClick = onBackClick
+            )
+        }
+    ) { paddingValues ->
+        AnimatedVisibility(visible = vm.showInternetCheckDialog) {
+            meteredDownloadConfirmationDialog(vm)
+        }
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(paddingValues)
+                .padding(vertical = 16.dp, horizontal = 24.dp)
+                .verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.spacedBy(32.dp)
+        ) {
+            Header(vm)
+            vm.changelog?.let { changelog ->
+                Divider()
+                Changelog(changelog)
+            } ?: Spacer(modifier = Modifier.weight(1f))
+            Buttons(vm, onBackClick)
+        }
+    }
+}
+
+@Composable
+private fun meteredDownloadConfirmationDialog(vm: UpdateViewModel) {
```

Comment:
This should be called `MeteredDownloadConfirmationDialog` because it returns `Unit`